jax/_src/numpy/lax_numpy.py

```diff
@@ -2369,6 +2369,11 @@ def tri(N, M=None, k=0, dtype=None):
   lax_internal._check_user_dtype_supported(dtype, "tri")
   M = M if M is not None else N
   dtype = dtype or float32
+  try:
+    k = operator.index(k)
+  except TypeError:
+    warnings.warn("using k supports int(k) but not k.__index__, e.g. float will be deprecated.", DeprecationWarning)
```

Comment:
I don't think the meaning or intent of this warning will be clear to users who see it. How about:
```
warnings.warn(f"jnp.tri: k must be an integer. In future JAX versions non-integer values will result in an error.")
```

Reply:
Thanks! I am not familiar with how to write a clear warning.